velox/functions/sparksql/tests/RegexFunctionsTest.cpp

```diff
@@ -41,6 +43,223 @@ class RegexFunctionsTest : public test::SparkFunctionBaseTest {
     return evaluateOnce<std::string>(
         fmt::format("regexp_extract(c0, '{}')", pattern), str);
   }
+
+  void testRegexReplaceSimpleOnce(
+      const std::vector<std::optional<std::string>>& input,
+      std::string pattern,
+      std::string replace,
+      std::string output,
+      std::optional<int> position =
+          std::nullopt /*adding the optional position parameter*/) {
+    auto valueAt = [&input](vector_size_t row) {
+      return input[row] ? StringView(*input[row]) : StringView();
+    };
+    auto nullAt = [&input](vector_size_t row) {
+      return !input[row].has_value();
+    };
+    auto result = [&] {
+      auto inputString =
+          makeFlatVector<StringView>(input.size(), valueAt, nullAt);
+      auto rowVector = makeRowVector({inputString});
+
+      if (!position) {
+        return evaluateOnce<std::string>(
+            fmt::format("regex_replace(c0, '{}', '{}')", pattern, replace),
+            rowVector);
+      } else {
+        return evaluateOnce<std::string>(
+            fmt::format(
+                "regex_replace(c0, '{}', '{}', {})",
+                pattern,
+                replace,
+                *position),
+            rowVector);
+      }
+    }();
+
+    EXPECT_EQ(result, output);
+  }
+
+  void testingRegexReplaceRowsSimple(
+      const std::vector<std::optional<std::string>>& input,
+      const std::vector<std::optional<std::string>>& pattern,
+      const std::vector<std::optional<std::string>>& replace,
+      const std::vector<std::optional<std::string>>& output,
+      const std::optional<std::vector<int64_t>>& positionOpt = std::nullopt,
+      int repeatCount = 1) {
+    if (repeatCount < 1) {
```

Comment:
Better to assert that repeatCount >= 1. Otherwise, folks may write tests that do nothing without realizing that.

Reply:
Fixed with EXPECT_GT(repeatCount, 0) in new commit

Reply:
Had to make EXPECT because ASSERT caused compile error since it does not match function return statement